### PR TITLE
Added ApplyMotionSetComponent

### PR DESCRIPTION
**Pull request by alexeykostin**
## What does this PR do?

Added missed ApplyMotionSetComponent

## How was this PR tested?

Tested locally in the Editor

### CODE REVIEW

File: Gems/EMotionFX/Code/Include/Integration/ApplyMotionSetComponentBus.h [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+#include <AzCore/Math/Vector3.h>
+#include <AzCore/Math/Quaternion.h>
+#include <AzCore/Asset/AssetCommon.h>
+#include <AzCore/Component/EntityId.h>
+
+namespace EMotionFX
+{
+    namespace Integration
+    {
+#if defined(CARBONATED)
```
Comment (vmednonogov) [inline Gems/EMotionFX/Code/Include/Integration/ApplyMotionSetComponentBus.h:24]: Looks like inner #ifdef/#endif  for CARBONATED are not necessary due to we use #if defined(CARBONATED) for whole file in the line 3
?

File: Gems/EMotionFX/Code/Include/Integration/ApplyMotionSetComponentBus.h [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+         */
+        class ApplyMotionSetComponentRequests
+            : public AZ::ComponentBus
+        {
+        public:
+
+            static const AZ::EBusHandlerPolicy HandlerPolicy = AZ::EBusHandlerPolicy::Single;
+
+            // applies the defined motion set to this Entity's AnimGraphInstance
+#if defined(CARBONATED)
```
Comment (vmednonogov) [inline Gems/EMotionFX/Code/Include/Integration/ApplyMotionSetComponentBus.h:46]: Looks like inner #ifdef/#endif  for CARBONATED are not necessary due to we use #if defined(CARBONATED) for whole file in the line 3
?

File: Gems/EMotionFX/Code/Include/Integration/ApplyMotionSetComponentBus.h [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+            virtual void Apply(const AZ::EntityId& id, const MotionSetGender& preferredGender) = 0;
+#else
```
Comment (vmednonogov) [inline Gems/EMotionFX/Code/Include/Integration/ApplyMotionSetComponentBus.h:48]: #else will never work ?

File: Gems/EMotionFX/Code/Include/Integration/ApplyMotionSetComponentBus.h [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+        using ApplyMotionSetComponentRequestBus = AZ::EBus<ApplyMotionSetComponentRequests>;
+
+        //Editor
+        class EditorApplyMotionSetComponentRequests
+            : public AZ::ComponentBus
+        {
+        public:
+            static const AZ::EBusHandlerPolicy HandlerPolicy = AZ::EBusHandlerPolicy::Single;
+            /// Retrieves the component's MotionSet AssetId.
+            /// \return assetId
```
Comment (vmednonogov) [inline Gems/EMotionFX/Code/Include/Integration/ApplyMotionSetComponentBus.h:62]: \return -> return

File: Gems/EMotionFX/Code/Source/Integration/Components/ApplyMotionSetComponent.cpp [hunk 5]
```diff
@@ -0,0 +1,187 @@
...
+    namespace Integration
+    {
+        //////////////////////////////////////////////////////////////////////////
+        void ApplyMotionSetComponent::Configuration::Reflect(AZ::ReflectContext* context)
+        {
+            auto* serializeContext = azrtti_cast<AZ::SerializeContext*>(context);
+            if (serializeContext)
+            {
+                serializeContext->Class<Configuration>()
+#if defined(CARBONATED)
```
Comment (vmednonogov) [inline Gems/EMotionFX/Code/Source/Integration/Components/ApplyMotionSetComponent.cpp:34]: Looks like inner #ifdef/#endif for CARBONATED are not necessary 
here and below
due to we use #if defined(CARBONATED) for whole file in the line 1
?

File: Gems/EMotionFX/Code/Source/Integration/Components/ApplyMotionSetComponent.h [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+            /**
+            * Configuration struct for procedural configuration of Actor Components.
+            */
+            struct Configuration
+            {
+                AZ_TYPE_INFO(Configuration, "{C1DD0FAF-0DEA-4965-940A-0E8A3FE8EABD}")
+
+                Configuration();
+
+#if defined(CARBONATED)
```
Comment (vmednonogov) [inline Gems/EMotionFX/Code/Source/Integration/Components/ApplyMotionSetComponent.h:47]: Looks like inner #ifdef/#endif for CARBONATED are not necessary
here and below
due to we use #if defined(CARBONATED) for whole file in the line 3
?

File: Gems/EMotionFX/Code/Source/Integration/Editor/Components/EditorApplyMotionSetComponent.cpp [hunk 7]
```diff
@@ -0,0 +1,324 @@
...
+    namespace Integration
+    {
+        //////////////////////////////////////////////////////////////////////////
+        void EditorApplyMotionSetComponent::Reflect(AZ::ReflectContext* context)
+        {
+            auto* serializeContext = azrtti_cast<AZ::SerializeContext*>(context);
+            if (serializeContext)
+            {
+                serializeContext->Class<EditorApplyMotionSetComponent, AzToolsFramework::Components::EditorComponentBase>()
+#if defined(CARBONATED)
```
Comment (vmednonogov) [inline Gems/EMotionFX/Code/Source/Integration/Editor/Components/EditorApplyMotionSetComponent.cpp:37]: Looks like inner #ifdef/#endif for CARBONATED are not necessary
here and below
due to we use #if defined(CARBONATED) for whole file in the line 14
?

File: Gems/EMotionFX/Code/Source/Integration/Editor/Components/EditorApplyMotionSetComponent.h [hunk 8]
```diff
@@ -0,0 +1,99 @@
...
+            static void GetRequiredServices(AZ::ComponentDescriptor::DependencyArrayType& required)
+            {
+                ApplyMotionSetComponent::GetRequiredServices(required);
+            }
+
+            static void Reflect(AZ::ReflectContext* context);
+            //////////////////////////////////////////////////////////////////////////
+
+        private:
+#if defined(CARBONATED)
```
Comment (vmednonogov) [inline Gems/EMotionFX/Code/Source/Integration/Editor/Components/EditorApplyMotionSetComponent.h:75]: Looks like inner #ifdef/#endif for CARBONATED are not necessary
here and below
due to we use #if defined(CARBONATED) for whole file in the line 3
?

File: Gems/EMotionFX/Code/Include/Integration/ApplyMotionSetComponentBus.h [hunk 9]
```diff
@@ -0,0 +1,61 @@
+#pragma once
```
Comment (vmednonogov) [inline Gems/EMotionFX/Code/Include/Integration/ApplyMotionSetComponentBus.h:9]: Validator fails for all files H and CPP in your PR
with message
**Validation / build (pull_request)
All checks have failed**

Comment (vmednonogov) [reply Gems/EMotionFX/Code/Include/Integration/ApplyMotionSetComponentBus.h:9]: Alexander O. fixed this issue ^^^ for his PR in O3DE fork some time ago.

File: Gems/EMotionFX/Code/Include/Integration/ApplyMotionSetComponentBus.h (hunk 4 above)
Comment (alexeykostin) [reply Gems/EMotionFX/Code/Include/Integration/ApplyMotionSetComponentBus.h:62]: This is likely a Doxygen markup

File: Gems/EMotionFX/Code/Include/Integration/ApplyMotionSetComponentBus.h (hunk 1 above)
Comment (alexeykostin) [reply Gems/EMotionFX/Code/Include/Integration/ApplyMotionSetComponentBus.h:24]: I just wanted to keep it closer to the LY source. If you wish I can refactor it.

File: Gems/EMotionFX/Code/Include/Integration/ApplyMotionSetComponentBus.h (hunk 9 above)
Comment (alexeykostin) [reply Gems/EMotionFX/Code/Include/Integration/ApplyMotionSetComponentBus.h:9]: These are tabs. The validator dislikes them. Deleted.

**Pull request merged**